pandas/core/reshape/merge.py

```diff
@@ -130,6 +130,9 @@
 if np.intc is not np.int32:
     _factorizers[np.intc] = libhashtable.Int64Factorizer
 
+if np.uintc is not np.uint32:
+    _factorizers[np.uintc] = libhashtable.UInt32Factorizer
```

Comment:
What is uintc a typedef for? This seems strange that we check if it is not uint32 but then still use the UInt32Factorizer anyway

Reply:
there should be UInt64Factorizer right ???

Reply:
Probably. Its hard to say for sure though without knowing what np.uintc is supposed to be